hyperspy/tests/io/test_bcf.py

```diff
@@ -194,3 +194,17 @@ def test_wrong_file():
     filename = os.path.join(my_path, 'bcf_data', 'Nope.bcf')
     with pytest.raises(TypeError):
         load(filename)
+
+def test_decimal_regex():
+    lxml = pytest.importorskip("lxml")
+    from hyperspy.io_plugins.bcf import fix_dec_patterns
+    dummy_xml_positive = [b'<dummy_tag>85,658</dummy_tag>',
+                          b'<dummy_tag>85,658E-8</dummy_tag>',
+                          b'<dummy_tag>-85,658E-8</dummy_tag>']
+    dummy_xml_negative = [b'<dum_tag>12,25,23,45,56,12,45</dum_tag>',
+                          b'<dum_tag>12e1,23,-24E-5</dum_tag>']
+    for i in dummy_xml_positive:
+        assert b'85.658' in fix_dec_patterns.sub(b'\\1.\\2', i)
+    for j in dummy_xml_negative:
+        assert b'.' not in fix_dec_patterns.sub(b'\\1.\\2', j)
+    
```

Comment:
Add a "negative control", i.e. a test to check that a decimal number originally formatted as `85.658` stays as `85.658`?

Reply:
Negative control had been added